components/Tabs/SettingTab.tsx

```diff
@@ -1,16 +1,30 @@
+import React, { useState } from 'react';
 import {View, Text, TouchableOpacity, StyleSheet} from 'react-native';
 
-const SettingTab = () => (
-  <View style={styles.container}>
-    <Text>Current Key</Text>
-    <TouchableOpacity
-      style={styles.actionButton}
-    >
-      <Text style={styles.actionText}>Generate Key</Text>
-    </TouchableOpacity>    
-  </View>
-);
 
+const SettingTab = () => {
+  const [generatedKey, setGeneratedKey] = useState(null);
```

Comment:
I think it would be best if you define the shape/type of the generatedKey, so instead of useState(null). Maybe you could make it useState<number | null>(null)